pytensor/graph/basic.py

```diff
@@ -1128,6 +1128,8 @@ def truncated_graph_inputs(
             else:
                 # otherwise, do not search beyond
                 truncated_inputs.append(node)
+        # add node to seen, no point in checking it once more
+        seen.add(node)
```

Comment:
We shouldn't call these nodes, as that's usually what we call Apply objects. They are variables in PyTensor slang.

Also should avoid adding already seen "nodes" to the queue as well?

Reply:
> We shouldn't call these nodes
Good idea, I checked it does not break anything outside the function and did the renaming

> Also should avoid adding already seen "nodes" to the queue as well?
Just added